File: C#/SymmetricDifference/SymmetricDifference.cs
using System.Collections.Generic;
using System.Linq;

namespace SymmetricDifferenceAlgorithm
{
    internal class SymmetricDifference
    {
        private int[][] _SETS;
        private List<int> _SYMMETRIC_DIFFERENCE;

        public SymmetricDifference(int[][] sets)
        {
            this._SETS = sets;
            this._SYMMETRIC_DIFFERENCE = _SETS[0].ToList();
        }

        public List<int> Calculate()
        {
            for (int i = 1; i < _SETS.Length; i++)
            {
                _SYMMETRIC_DIFFERENCE = Difference(_SYMMETRIC_DIFFERENCE, _SETS[i]);
            }

            return _SYMMETRIC_DIFFERENCE;
        }

        private bool IsUnique(int control, List<int> symmetric_difference)
        {
            if (symmetric_difference.Contains(control))
            {
                return false;
            }
            else
            {
                return true;
            }
        }

        private List<int> Difference(List<int> set1, int[] set2)
        {
            List<int> symmetric_difference = set1.AsQueryable().Distinct().ToList();
            List<int> group = set2.ToList().AsQueryable().Distinct().ToList();

            for (int i = 0; i < group.Count; i++)
            {
                if (IsUnique(group[i], symmetric_difference))
                {
                    symmetric_difference.Add(group[i]);
                }
                else
                {
                    symmetric_difference.Remove(group[i]);
                }
            }

            symmetric_difference.Sort();
            return symmetric_difference;
        }
    }
}

File: JavaScript/symmetric-difference.js
function sortArray(...args)
{
    if(args[0] === args[1])
    {
        return 0;
    }
    else
    {
        return (args[0] < args[1]) ? -1 : 1;
    }
}

function isUnique(control, group)
{
    if (group.includes(control))
    {
        return false;
    }
    else
    {
        return true;
    }
}

function symmetricDifference(...args)
{
    let symmetric_difference = [...new Set(args[0])];
    args[1] = [...new Set(args[1])];
    
    for(let i = 0; i < args[1].length; i++)
    {
        if(isUnique(args[1][i], symmetric_difference))
        {
            symmetric_difference.push(args[1][i]);
        }
        else
        {
            symmetric_difference = symmetric_difference.filter(num => num !== args[1][i]).slice();
        }
    }

    return symmetric_difference.sort(sortArray);
}

function sym(...args)
{
    let symmetric_difference = args[0];

    for (let i = 1; i < args.length; i++)
    {
        symmetric_difference = symmetricDifference(symmetric_difference, args[i]).slice();
    }

    return symmetric_difference;
}

console.log(sym([1, 2, 3], [5, 2, 1, 4, 5], [0, 9, 15, 1, 4, 81]));


